qemu/lib/iozone.py

```diff
@@ -0,0 +1,164 @@
+        logging.info("Write results to %s" % self.results_path)
+        with open(self.results_path, 'w') as file:
+            file.write(output)
+
+        return status, output
+
+    def after_run_iozone(self):
+        """
+        Clean up iozone env/post result after iozone test
+        """
+        pass
+
+
+class Iozone_Win(IozoneTest):
+    """
+    Check driver before iozone test, post result after iozone test.
+    """
+    def __init__(self, vm, params, test):
+        IozoneTest.__init__(self, vm, params, test)
+
+    def get_driver(self):
+        """
+        Get driver name according to driver_format
+        """
+        driver_name = self.params.get("driver_name", "")
+        drive_format = self.params.get("drive_format")
+        if not driver_name:
+            if "scsi" in drive_format:
+                driver_name = "vioscsi"
+            elif "virtio" in drive_format:
+                driver_name = "viostor"
+            else:
+                driver_name = None
+        return driver_name
+
+    def driver_verify(self):
+        """
+        Run driver verification if needed
+        """
+        driver_name = self.get_driver()
+        if driver_name:
+            utils_test.qemu.windrv_check_running_verifier(self.session, self.vm,
+                                                          self.test, driver_name)
+
+    def before_run_iozone(self):
+        """
+        Do driver verification before iozone test
+        """
+        logging.info("Do driver verification for virtio")
+        self.driver_verify()
+
+    def after_run_iozone(self):
+        """
+        Post result after iozone test, generate a series graphs
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -b
+        # I:\iozone.xls -f I:\testfile".
+        # Otherwise, your result can't be analysed correctly.
+        if self.params.get("post_results", "no") == "yes":
+            logging.info("Generate graph of test result")
+            analysisdir = os.path.join(self.test.resultsdir, 'analysis_%s'
+                                       % self.test.iteration)
+            a = postprocess_iozone.IOzoneAnalyzer(list_files=[self.results_path],
+                                                  output_dir=analysisdir)
+            a.analyze()
+            p = postprocess_iozone.IOzonePlotter(results_file=self.results_path,
+                                                 output_dir=analysisdir)
+            p.plot_all()
+
+
+class Iozone_Linux(IozoneTest):
+    """
+    Install iozone test tool before test, clean up iozone packets after test
+    """
+    def __init__(self, vm, params, test):
+        IozoneTest.__init__(self, vm, params, test)
+        stress_type = "iozone"
+        self.iozone_work_path = params.get("iozone_work_path", "/home/iozone3_458/src/current")
+        download_url = params.get("download_url_iozone",
```

Comment:
I think `self.iozone_work_path` and `download_url` should be compulsory params

Reply:
These two params won't be compulsory,  suggest to use the default value to run test, or there will be code redundancy issue.

Only if a new iozone version needed in test, we will define it in our cfgs.